app/models/custom_attribute.rb

```diff
@@ -4,6 +4,8 @@ class CustomAttribute < ApplicationRecord
   belongs_to :resource, :polymorphic => true
   serialize :serialized_value
 
+  validates :name, :format => {:with => /\A[\p{Alpha}_][\p{Alpha}_\d\$]*\z/, :message => "must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters can be letters, underscores, digits (0-9), or dollar signs ($)"}
```

Comment:
I wish validation messages showed the tested string, it's not clear what failed in the tests... you have to go to the test and find what name we're providing..."

```
Validation failed: Name must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters can be letters, underscores, digits (0-9), or dollar signs ($)
```